drivers/spi/spi.go

```diff
@@ -0,0 +1,44 @@
+package spi
+
+import (
+	"golang.org/x/exp/io/spi"
+	"sync"
+	"golang.org/x/exp/io/spi/driver"
+	"fmt"
+)
+
+type Connection struct {
+	connection *spi.Device
+	address int
+	mutex *sync.Mutex
+}
+
+// NewConnection creates and returns a new connection to a specific
+// spi device on a bus and address
+func NewConnection(opener driver.Opener, address int) (connection *Connection) {
+	dev, err := spi.Open(opener)
+	if err != nil {
+		panic(err)
+	}
+
+	return &Connection{connection: dev, address: address, mutex: &sync.Mutex{}}
+}
+
+// TODO: Read - the exp/io/spi implementation doesn't currently support reading
```

Comment:
Also, some changes to the Makefile https://github.com/hybridgroup/gobot/blob/master/Makefile#L44 are going to be needed to this branch, in order to allow the test suite to run.

Please let me know anything I can do to help out!